xarray/core/dataset.py

```diff
@@ -709,6 +709,7 @@ def _replace(  # type: ignore
         indexes: 'Optional[OrderedDict[Any, pd.Index]]' = __default,
         encoding: Optional[dict] = __default,
         inplace: bool = False,
+        deep: bool = False,
```

Comment:
Rather than modifying `_replace`, could you use pass in `attrs` explicitly from `Dataset.copy()` itself?

I'd like to keep this method relatively simple -- it already has a large number of keyword arguments.